visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/BusinessLayer/VisitServiceImpl.java

```diff
@@ -15,33 +15,42 @@
 public class VisitServiceImpl implements VisitService {
     private final VisitRepo repo;
 
-
     @Override
-    public Mono<VisitResponseDTO> addVisit(Mono<VisitRequestDTO> visitRequestDTOMono) {
-        return visitRequestDTOMono
-                .map(EntityDtoUtil::toVisitEntity)
-                .doOnNext(x -> x.setVisitId(EntityDtoUtil.generateVisitIdString()))
-                .flatMap((repo::insert))
-                .map(EntityDtoUtil::toVisitResponseDTO);
+    public Flux<VisitResponseDTO> getAllVisits(){
+        return repo.findAll().map(EntityDtoUtil::toVisitResponseDTO);
     }
-
     @Override
     public Flux<VisitResponseDTO> getVisitsForPet(int petId) {
         return repo.findByPetId(petId)
                 .map(EntityDtoUtil::toVisitResponseDTO);
     }
-
+    @Override
+    public Flux<VisitResponseDTO> getVisitsForPractitioner(int practitionerId) {
+        return repo.findVisitsByPractitionerId(practitionerId)
+                .map(EntityDtoUtil::toVisitResponseDTO);
+    }
+    @Override
+    public Flux<VisitResponseDTO> getVisitsByPractitionerIdAndMonth(int practitionerId, int month) {
+        return repo.findVisitsByPractitionerIdAndMonth(practitionerId, month)
+                .map(EntityDtoUtil::toVisitResponseDTO);
+    }
     @Override
     public Mono<VisitResponseDTO> getVisitByVisitId(String visitId) {
         return repo.findByVisitId(visitId)
                 .map(EntityDtoUtil::toVisitResponseDTO);
     }
-
+    @Override
+    public Mono<VisitResponseDTO> addVisit(Mono<VisitRequestDTO> visitRequestDTOMono) {
+        return visitRequestDTOMono
+                .map(EntityDtoUtil::toVisitEntity)
+                .doOnNext(x -> x.setVisitId(EntityDtoUtil.generateVisitIdString()))
+                .flatMap((repo::insert))
+                .map(EntityDtoUtil::toVisitResponseDTO);
+    }
```

Comment:
Method is reactive, was moved around, could check exceptions, could set the visitId in the toVisitEntity method, looks correct.